pkg/sql/codegen/pai/template_tf.go

```diff
@@ -0,0 +1,110 @@
+// Copyright 2019 The SQLFlow Authors. All rights reserved.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package pai
+
+type wrapperFiller struct {
+	DataSource string
+	EntryFile  string
+	ModelName  string
+}
+
+type saveModelFiller struct {
+	DataSource   string
+	ModelName    string
+	Estimator    string
+	Save         string
+	IsKerasModel bool
+}
+
+type predictFiller struct {
+	DataSource  string
+	ModelName   string
+	Select      string
+	ResultTable string
+}
+
+const tfWrapperTmplText = `
+import os
+import subprocess
+import tarfile
+
+import sqlflow_submitter.db
+
+tarball = "/%s/sqlflow_model.tar.gz" % os.getcwd()
+archive = tarfile.open(tarball, "w|gz")
+
+# '.' is always in sys.path
+archive.add(sqlflow_submitter.__path__[0], arcname='sqlflow_submitter')
+archive.add('{{.EntryFile}}')
+archive.close()
+
+driver, dsn = "{{.DataSource}}".split("://")
+assert driver == "maxcompute"
+user, passwd, address, database = sqlflow_submitter.db.parseMaxComputeDSN(dsn)
+
+pai_cmd = 'pai -name %s -DjobName=%s -Dtags=%s -Dscript=file://%s -DentryFile=%s' % (
+	'tensorflow1120', 'sqlflow_' + '{{.ModelName}}', 'dnn', tarball, '{{.EntryFile}}')
+
+# Submit the tarball to PAI
+subprocess.run(["odpscmd", "-u", user,
```

Comment:
Need to check the return code in case that the command fails.

Reply:
Done.